transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java

```diff
@@ -771,6 +773,28 @@ public V apply(final K k) {
     private final Map<Store<?, ?>, SoftLockValueCombinedSerializerLifecycleHelper> createdStores = new ConcurrentWeakIdentityHashMap<Store<?, ?>, SoftLockValueCombinedSerializerLifecycleHelper>();
 
     @Override
+    public int rank(final Set<ResourceType> resourceTypes, final Collection<ServiceConfiguration<?>> serviceConfigs) {
+      if (findSingletonAmongst(XAStoreConfiguration.class, serviceConfigs) == null) {
+        // An XAStore must be configured for use
+        return 0;
+      }
+
+      // TODO: Introduce proper Provider discovery & tracking and remove this *HACK*
+      final int underlyingRank;
+      if (resourceTypes.equals(EnumSet.of(ResourceType.Core.HEAP))) {
```

Comment:
I used this hack instead of simply presuming the underlying provider (`DefaultStoreProvider`) supported the configuration.  This will need changing once `XSStore.Provider` is capable of discovering (and tracking) the proper provider.

(As a note, `DefaultStoreProvider` continues to support heap-only -- just doesn't "advertise" that fact.)